src/Symfony/Component/Serializer/Mapping/Factory/CacheClassMetadataFactory.php

```diff
@@ -51,14 +51,14 @@ public function getMetadataFor($value)
             return $this->loadedClasses[$class];
         }
 
-        $key = rawurlencode(strtr($class, '\\', '_'));
+        $metadata = $this->decorated->getMetadataFor($value);
```

Comment:
Isn't this basically circumventing the cache now that the metadata is loaded here?

Reply:
Maybe we can switch to another implementation of debug is enabled on the framework bundle extension. Need some help here so we can fix this issue that only happens in dev.